File: service/app/src/test/java/io/miragon/blueprint/domain/leasing/CustomerNameTest.java
package io.miragon.blueprint.domain.leasing;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.Test;

class CustomerNameTest {

    @Test
    void exposesTheWrappedName() {
        // given/when: a customer name is created from a non-blank value
        CustomerName name = new CustomerName("John Doe");
        // then: the raw value is exposed unchanged
        assertThat(name.value()).isEqualTo("John Doe");
    }

    @Test
    void rejectsAnEmptyName() {
        // when/then: an empty value is refused
        assertThatThrownBy(() -> new CustomerName(""))
                .isInstanceOf(IllegalArgumentException.class);
    }

    @Test
    void rejectsABlankName() {
        // when/then: a whitespace-only value is refused
        assertThatThrownBy(() -> new CustomerName("   "))
                .isInstanceOf(IllegalArgumentException.class);
    }
}

File: service/app/src/test/kotlin/io/miragon/blueprint/domain/leasing/CustomerNameTest.kt
package io.miragon.blueprint.domain.leasing

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class CustomerNameTest {

    @Test
    fun `exposes the wrapped name`() {
        // given/when: a customer name is created from a non-blank value
        val name = CustomerName("John Doe")
        // then: the raw value is exposed unchanged
        assertThat(name.value).isEqualTo("John Doe")
    }

    @Test
    fun `rejects an empty name`() {
        // when/then: an empty value is refused
        assertThatThrownBy { CustomerName("") }
            .isInstanceOf(IllegalArgumentException::class.java)
    }

    @Test
    fun `rejects a blank name`() {
        // when/then: a whitespace-only value is refused
        assertThatThrownBy { CustomerName("   ") }
            .isInstanceOf(IllegalArgumentException::class.java)
    }
}
